tests-e2e/cypress/integration/runs/retrospective_spec.js

```diff
@@ -49,16 +49,26 @@ describe('runs > retrospective', () => {
         cy.visit(`/playbooks/runs/${runId}/retrospective`);
 
         // # Start editing
-        cy.findByText('Edit').click();
+        cy.findByTestId('retro-report-text').click();
 
         // * Verify the provided template text is pre-filled
         cy.focused().should('include.text', 'This is a retrospective template.');
 
-        // # Change the retro text, save it, and publish
+        // # Change the retro text
         cy.focused().clear().type('Edited retrospective.');
-        cy.findByText('Save').click();
+
+        // # Save it by clicking outside the text area
+        cy.findByText('Report').click();
+
+        // # Publish
         cy.findByText('Publish').click({force: true});
 
+        // * Verify we're showing the publish retro confirmation modal
+        cy.get('#confirmModalLabel').contains('Publish retrospective');
+
+        // # Publish
+        cy.get('#confirmModalButton').click({force: true});
```

Comment:
And similarly here, we could probably look for the Publish button (maybe `within` the dialog, if needed)
```suggestion
        cy.findByRole('button', {name: 'Publish'}).click({force: true});
```

Reply:
Similar here --not sure why we should bother if the id is already there? Personally, I think it's a poor use of our effort to refactor the tests in this way? (2/5)

Reply:
I agree that refactoring existing tests that already work just for the sake of making them assess accessibility levels is not worth it. But when we write new tests, or are already modifying ones, I think it may be a good practice to use these queries so that we don't only test the feature, but the accessibility itself: if it's hard to make a query, maybe there's room for improvement for making the site more accessible.

That said, in this specific example we may not need it, so 0/5 on changing this line of code. Having this conversation was already valuable :nerd_face: